sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Customization/Models/RestorableCosmosDBAccountResource.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.CosmosDB.Models;
+
+namespace Azure.ResourceManager.CosmosDB
+{
+    /// <summary>
+    /// A Class representing a RestorableCosmosDBAccount along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="RestorableCosmosDBAccountResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetRestorableCosmosDBAccountResource method.
+    /// Otherwise you can get one from its parent resource <see cref="CosmosDBLocationResource" /> using the GetRestorableCosmosDBAccount method.
+    /// </summary>
+    public partial class RestorableCosmosDBAccountResource : ArmResource
+    {
+        /// <summary>
+        /// Show the event feed of all mutations done on all the Azure Cosmos DB MongoDB collections under a specific database.  This helps in scenario where container was accidentally deleted.  This API requires &apos;Microsoft.DocumentDB/locations/restorableDatabaseAccounts/.../read&apos; permission
+        /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableMongodbCollections
+        /// Operation Id: RestorableMongodbCollections_List
+        /// </summary>
+        /// <param name="restorableMongoDBDatabaseRid"> The resource ID of the MongoDB database. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns> An async collection of <see cref="RestorableMongoDBCollection" /> that may take multiple service requests to iterate over. </returns>
+        public virtual AsyncPageable<RestorableMongoDBCollection> GetRestorableMongoDBCollectionsAsync(string restorableMongoDBDatabaseRid, CancellationToken cancellationToken)
```

Comment:
Why we have to introduce this customization file?
What issue do we have in the generated version of this method?

Reply:
@ArcturusZhang, we added 2 optional parameters to mongo collection restore API, that introduced a binary breaking change. We were facing issue with building dotnet sdk using latest swager spec. Thus, we added this customization to support these new optional parameters.